internal/services/synapse/synapse_spark_pool_resource.go

```diff
@@ -213,12 +213,13 @@ func resourceSynapseSparkPool() *pluginsdk.Resource {
 			"spark_version": {
 				Type:     pluginsdk.TypeString,
 				Optional: true,
-				Default:  "2.4",
+				Default:  "3.4",
```

Comment:
a more future-proof solution would be:
```suggestion
				Default:  "2.4", // TODO: make this field Required in 4.0 since the default value changes over time
```

Reply:
I agree.  This probably should be a required field.  Speaking from the experience of building a Synapse environment using the provider, we went down a bad path by allowing it to use the default.  And we had to revise our spark pool version later because our data engineer couldn't use the version of Python he didn't to.